File: android/lca/src/main/java/com/dreampany/lca/data/source/room/RoomIcoDataSource.java
/*
package com.dreampany.lca.data.source.room;

import com.dreampany.lca.data.enums.IcoStatus;
import com.dreampany.lca.data.misc.IcoMapper;
import com.dreampany.lca.data.model.Ico;
import com.dreampany.lca.data.source.api.IcoDataSource;

import java.util.List;

import javax.inject.Singleton;

import com.dreampany.lca.data.source.dao.IcoDao;

import io.reactivex.Maybe;

*/
/**
 * Created by Hawladar Roman on 7/10/2018.
 * BJIT Group
 * hawladar.roman@bjitgroup.com
 *//*


@Singleton
public class RoomIcoDataSource implements IcoDataSource {

    private final IcoMapper mapper;
    private final IcoDao dao;

    public RoomIcoDataSource(IcoMapper mapper,
                             IcoDao dao) {
        this.mapper = mapper;
        this.dao = dao;
    }

    @Override
    public boolean isEmpty() {
        return getCount() == 0;
    }

    @Override
    public Maybe<Boolean> isEmptyRx() {
        return Maybe.fromCallable(this::isEmpty);
    }

    @Override
    public int getCount() {
        return dao.getCount();
    }

    @Override
    public Maybe<Integer> getCountRx() {
        return null;
    }

    @Override
    public boolean isExists(Ico ico) {
        return false;
    }

    @Override
    public Maybe<Boolean> isExistsRx(Ico ico) {
        return null;
    }

    @Override
    public long putItem(Ico ico) {
        return dao.insertOrReplace(ico);
    }

    @Override
    public Maybe<Long> putItemRx(Ico ico) {
        return Maybe.fromCallable(() -> putItem(ico));
    }

    @Override
    public List<Long> putItems(List<?extends Ico> icos) {
        return dao.insertOrReplace(icos);
    }

    @Override
    public Maybe<List<Long>> putItemsRx(List<? extends Ico> icos) {
        return Maybe.fromCallable(() -> putItems(icos));
    }

    @Override
    public int delete(Ico ico) {
        return 0;
    }

    @Override
    public Maybe<Integer> deleteRx(Ico ico) {
        return null;
    }

    @Override
    public List<Long> delete(List<? extends Ico> icos) {
        return null;
    }

    @Override
    public Maybe<List<Long>> deleteRx(List<? extends Ico> icos) {
        return null;
    }

    @Override
    public Ico getItem(String id) {
        return null;
    }

    @Override
    public Maybe<Ico> getItemRx(String id) {
        return null;
    }

    @Override
    public List<Ico> getItems() {
        return null;
    }

    @Override
    public Maybe<List<Ico>> getItemsRx() {
        return null;
    }

    @Override
    public List<Ico> getItems(int limit) {
        return null;
    }

    @Override
    public Maybe<List<Ico>> getItemsRx(int limit) {
        return null;
    }

*/
/*    @Override
    public void clear(IcoStatus status) {
        mapper.clear(status);
    }*//*



    @Override
    public List<Ico> getLiveItems(int limit) {
        return null;
    }

    @Override
    public Maybe<List<Ico>> getLiveItemsRx(int limit) {
        return dao.getItemsRx(IcoStatus.LIVE.name(), limit);
    }

    @Override
    public List<Ico> getUpcomingItems(int limit) {
        return null;
    }

    @Override
    public Maybe<List<Ico>> getUpcomingItemsRx(int limit) {
        return dao.getItemsRx(IcoStatus.UPCOMING.name(), limit);
    }

    @Override
    public List<Ico> getFinishedItems(int limit) {
        return null;
    }

    @Override
    public Maybe<List<Ico>> getFinishedItemsRx(int limit) {
        return dao.getItemsRx(IcoStatus.FINISHED.name(), limit);
    }
}
*/

File: android/lca/src/main/kotlin/com/dreampany/lca/data/source/room/RoomIcoDataSource.kt
package com.dreampany.lca.data.source.room

import com.dreampany.lca.data.enums.IcoStatus
import com.dreampany.lca.data.misc.IcoMapper
import com.dreampany.lca.data.model.Ico
import com.dreampany.lca.data.source.api.IcoDataSource
import com.dreampany.lca.data.source.dao.IcoDao
import io.reactivex.Maybe
import javax.inject.Singleton

/**
 * Created by roman on 2019-07-14
 * Copyright (c) 2019 bjit. All rights reserved.
 * hawladar.roman@bjitgroup.com
 * Last modified $file.lastModified
 */
@Singleton
class RoomIcoDataSource constructor(
    val mapper: IcoMapper,
    val dao: IcoDao
) : IcoDataSource{

    override fun isEmpty(): Boolean {
        return getCount() == 0
    }

    override fun isEmptyRx(): Maybe<Boolean> {
        return Maybe.fromCallable({ this.isEmpty() })
    }

    override fun getCount(): Int {
        return dao.count
    }

    override fun getCountRx(): Maybe<Int> {
        return Maybe.empty()
    }

    override fun isExists(ico: Ico): Boolean {
        return false
    }

    override fun isExistsRx(ico: Ico): Maybe<Boolean> {
        return Maybe.empty()
    }

    override fun putItem(ico: Ico): Long {
        return dao.insertOrReplace(ico)
    }

    override fun putItemRx(ico: Ico): Maybe<Long> {
        return Maybe.fromCallable { putItem(ico) }
    }

    override fun putItems(icos: List<Ico>): List<Long>? {
        return dao.insertOrReplace(icos)
    }

    override fun putItemsRx(icos: List<Ico>): Maybe<List<Long>> {
        return Maybe.fromCallable { putItems(icos) }
    }

    override fun delete(ico: Ico): Int {
        return 0
    }

    override fun deleteRx(ico: Ico): Maybe<Int> {
        return Maybe.empty()
    }

    override fun delete(icos: List<Ico>): List<Long>? {
        return null
    }

    override fun deleteRx(icos: List<Ico>): Maybe<List<Long>> {
        return Maybe.empty()
    }

    override fun getItem(id: String): Ico? {
        return null
    }

    override fun getItemRx(id: String): Maybe<Ico> {
        return Maybe.empty()
    }

    override fun getItems(): List<Ico>? {
        return null
    }

    override fun getItemsRx(): Maybe<List<Ico>> {
        return Maybe.empty()
    }

    override fun getItems(limit: Int): List<Ico>? {
        return null
    }

    override fun getItemsRx(limit: Int): Maybe<List<Ico>> {
        return Maybe.empty()
    }

    override fun getLiveItems(limit: Int): List<Ico>? {
        return null
    }

    override fun getLiveItemsRx(limit: Int): Maybe<List<Ico>> {
        return dao.getItemsRx(IcoStatus.LIVE.name, limit)
    }

    override fun getUpcomingItems(limit: Int): List<Ico>? {
        return null
    }

    override fun getUpcomingItemsRx(limit: Int): Maybe<List<Ico>> {
        return dao.getItemsRx(IcoStatus.UPCOMING.name, limit)
    }

    override fun getFinishedItems(limit: Int): List<Ico>? {
        return null
    }

    override fun getFinishedItemsRx(limit: Int): Maybe<List<Ico>> {
        return dao.getItemsRx(IcoStatus.FINISHED.name, limit)
    }
}